external-crates/move/move-compiler/src/typing/translate.rs

```diff
@@ -1609,13 +1598,11 @@ fn exp_inner(context: &mut Context, sp!(eloc, ne_): N::Exp) -> T::Exp {
 fn loop_body(
     context: &mut Context,
     eloc: Loc,
+    name: Var,
     is_loop: bool,
     nloop: Box<N::Exp>,
 ) -> (bool, Type, Box<T::Exp>) {
-    let old_loop_info = context.enter_loop();
     let eloop = exp(context, nloop);
```

Comment:
To clean up the casing here a bit, I think we can set the break type to `Unit` when `!is_loop`, then the code for breaking with a value will just work out naturally with `while`?